src/mesh_element/Element_Definition.cc

```diff
@@ -5,240 +5,335 @@
+    Require(side_type_.size() == number_of_sides_);
+    for (unsigned i = 0; i < number_of_sides_; ++i)
     {
-        Require( static_cast<unsigned>(side_type_[i])<elem_defs.size() );
+        Require(static_cast<unsigned>(side_type_[i]) < elem_defs_.size());
     }
-#endif
-    Require( side_nodes_.size()==number_of_sides_ );
-    Require( node_loc_.size()==number_of_nodes_ );
-#if DBC & 1
-    for (unsigned i=0; i<number_of_sides_; ++i)
+    Require(side_nodes_.size() == number_of_sides_);
+    for (unsigned i = 0; i < number_of_sides_; ++i)
     {
-        Require( side_nodes_[i].size() ==
-                 elem_defs_[side_type_[i]].get_number_of_nodes() );
-        
-        for (unsigned j=0; j<side_nodes_[i].size(); ++j)
+        Require(side_nodes_[i].size() ==
+                elem_defs_[side_type_[i]].get_number_of_nodes());
+        for (unsigned j = 0; j < side_nodes_[i].size(); ++j)
         {
-            Require( static_cast<unsigned>(side_nodes_[i][j])
-                     < number_of_nodes_ );
-            
-            Require( elem_defs_[side_type_[i]].get_node_location(j) ==
-                     node_loc_[side_nodes_[i][j]] );
+            Require(static_cast<unsigned>(side_nodes_[i][j]) <
+                    number_of_nodes_);
         }
     }
-#endif
-
-     Ensure( invariant_satisfied() );
-
-     Ensure( get_type()==Element_Definition::POLYGON );
-     Ensure( get_name()==name_  );
-     Ensure( get_dimension()==dimension_  );
-     Ensure( get_number_of_nodes()==number_of_nodes_  );
-     Ensure( get_number_of_sides()==number_of_sides_  );
-#if DBC & 4
-     for (unsigned i=0; i<number_of_nodes; ++i)
-     {
-         Ensure( get_node_location(i)==node_loc_[i]  );
-     }
-     for (unsigned i=0; i<number_of_sides; ++i)
-     {
-//         Ensure( get_side_type(i)==elem_defs_[side_type_[i]]  );
-         Ensure( get_side_nodes(i)==side_nodes_[i]  );
-     }
-#endif
-
-//     number_of_face_nodes.resize(side_type.size());
-//     for (unsigned s=0; s < side_type.size(); ++s)
-//         number_of_face_nodes[s] = get_side_type(s).get_number_of_nodes();
-
-//     face_nodes.resize(side_nodes.size());
-
-//     for (unsigned s=0; s < face_nodes.size(); ++s)
-//     {
-//         std::vector<int> nodes(get_side_nodes(s)); 
-//         face_nodes[s].resize(nodes.size());
-
-//         for (unsigned n=0; n < nodes.size(); ++n)
-//             face_nodes[s][n] = nodes[n];
-//     }
+
+    Ensure(get_type() == Element_Definition::POLYGON ||
+           get_type() == Element_Definition::POLYHEDRON);
+    Ensure(get_name() == name_);
+    Ensure(get_dimension() == dimension_);
+    Ensure(get_number_of_nodes() == number_of_nodes_);
+    Ensure(get_number_of_sides() == number_of_sides_);
+
+    for (unsigned i = 0; i < number_of_sides; ++i)
+    {
+        Ensure(get_side_nodes(i) == side_nodes_[i]);
+    }
+
+    //---------------------------------------------------------------------------//
```

Comment:
First attempt at the polyhedron. I was preserving it because the same functionality is used
elsewhere now in capsaicin.